### ISSUE TITLE
`samtools view --regions-file` multi-threading issue

### ISSUE THREAD

**Issue by dennishendriksen** [samtools/samtools#1890]
Running `samtools 1.17` on `Linux 3.10.0-1160.80.1.el7.x86_64` with
```
samtools view \
--reference GCA_000001405.15_GRCh38_no_alt_analysis_set.fna \
--regions-file HG003_16.bed \
--bam --output vip_AshkenazimTrio_HG003.cram.bam \
--threads 4 \
vip_AshkenazimTrio_HG003.cram
```
can produce a `bam` file that can't be indexed with
```
samtools index vip_AshkenazimTrio_HG003.cram.bam
```
resulting in error:
```
[W::hts_itr_multi_cram] No index entry for region 25:0-175055
[W::hts_itr_multi_cram] No index entry for region 26:0-32032
<cut>
[W::hts_itr_multi_cram] No index entry for region 186:0-40191
[W::hts_itr_multi_cram] No index entry for region 187:0-36723
[E::hts_idx_push] Chromosome blocks not continuous
[E::sam_index] Read 'A00744:46:HV3C3DSXX:2:1521:3179:31266' with ref_name='chrX', ref_length=156040895, flags=83, pos=113453652 cannot be indexed
samtools index: failed to create index for "vip_AshkenazimTrio_HG003.cram.bam"
```
sorting the output .bam does allow the file to be indexed but the output contains an unexpected number of reads according to `idxstats`:
```
# input
<cut>
chrX	156040895	589	5
chrY	57227415	158	1
<cut>

# output
<cut>
chrX	156040895	739	8
chrY	57227415	316	2
<cut>
```
the error can be consistently reproduced on multiple compute clusters, but can also *not* be reproduced on one compute cluster. running the `view` process single threaded resolves the issue on all clusters.

if I understand the documentation correctly `--regions-file` removes duplicate sequences due to equivalence to `-M -L`? my assumption if that `samtools view` produces sorted output for sorted input. is this correct?

resources can be downloaded [here](https://downloads.molgeniscloud.org/downloads/vip_dev/github/431/).

**Comment by daviesrob** [comment]
Thanks for the bug report and data.  I've checked, and this was caused by a bug in HTSlib's cram index reading code, which was fixed by samtools/htslib#1574.  If you build the `develop` branch, the `samtools view` command will work correctly and `idxstats` will return the expected numbers.

This fix will be included in the 1.18 release, which should be out in the next few days.

**Comment by daviesrob** [comment]
We've now released 1.18, including the fix for this problem.
